NitroxClient/MonoBehaviours/Gui/HUD/RemotePlayerVitals.cs

```diff
@@ -211,10 +231,10 @@ public void SetNewPosition(int newPosition)
 
         private void SetBarPostion(GameObject barGameObject, Vector2 positionOffset, Canvas canvas)
         {
-            Vector2 screenPosition = new Vector2(Screen.width - positionOffset.x, Screen.height - (positionOffset.y * position));
-            Vector2 worldPosition;
-            RectTransformUtility.ScreenPointToLocalPointInRectangle(canvas.transform as RectTransform, screenPosition, Camera.main, out worldPosition);
-            barGameObject.transform.position = canvas.transform.TransformPoint(worldPosition);
+            //Vector2 screenPosition = new Vector2(Screen.width - positionOffset.x, Screen.height - (positionOffset.y * position));
+            //Vector2 worldPosition;
+            //RectTransformUtility.ScreenPointToLocalPointInRectangle(canvas.transform as RectTransform, screenPosition, Camera.main, out worldPosition);
+            barGameObject.transform.position = Camera.main.WorldToScreenPoint(PingManager.Get(pingId).transform.position) + new Vector3(positionOffset.x, positionOffset.y);
```

Comment:
Just use `+ (Vector3)positionOffset`, `Vector2` has an `implicit operator` defined for it (though you have to explicitly specify it, because the compiler is unsure whether to convert the result of `WorldToScreenPoint` from `Vector3` to `Vector2`, or `positionOffset` from `Vector2` to `Vector3` since both conversions are implicit).

Also, the `PingInstance` returned by `PingManager.Get` is the same as initially registered with `Register`, is there a reason to not store the `PingInstance` in a `private` field in this class?

Reply:
ok I see your logic behind this. My initial intention was to Register to the ping manager and then get the position of the UI ping inside the manager, but I couldn't see any plain functions that achieved this, so i am setting it to the signal position.

Reply:
You'll want to change this a little anyway, since it seems the position of the ping is always equal to that of the local player (when it was created initially), not of the player these bars are for.

Though that probably 'solves itself' as soon as you use the pings from `RemotePlayer` - those are attached to the player and thus have the right position.